### ISSUE TITLE
[Feature Request] Buttons on Popup

### ISSUE THREAD

**Issue by GreenLunar** [evilsocket/opensnitch#477]
Hello, I'm a happy ArchLinux user of OpenSnitch under Xfce desktop, and I want to ask for a new feature.
Following issue #476, I'd like to suggest to have an option to enable notification using a notification balloon.

The notification balloon would have 3 three buttons "Deny" "Allow" "Expert"
"Expert" button (or whatever name you have in mind) will open the popup dialog which we currently have.

**Reason: Intrusiveness**
While Notification Balloon will be on the top layer of the display, just as the current popup dialog, it will not be intrusive in the sense that it will not cause a fault of a current user's action.

**Example: Intrusiveness**
User is in the middle of adding tags to a bookmark in a web browser, and the popup dialog of OpenSnitch just caused it to be closed.

**Comment by gustavo-iniguez-goya** [comment]
Hi @GreenLunar ,

I like the idea of accepting/denying a connection from a notification, mainly because on some systems there's no systray area, and besides something is changing on this matter https://wiki.debian.org/Ayatana/IndicatorsTransition, https://lists.debian.org/debian-devel/2018/03/msg00506.html 

However I think that it should be configurable, users should choose what kind of pop-ups to display: normal (current pop-ups) or minimal (icons on the notification). We already talked about making the pop-ups smaller here #424 and I even prototyped a simpler pop-up, but I think this approach is better.

**Comment by GreenLunar** [comment]
Yes, making it configurable (optional) is a good idea.
P.S. The current popup is great.

**Comment by gustavo-iniguez-goya** [comment]
Partially done. On GTK based DEs works from Ubuntu >= 16. On KDE (and others probably) I need to figure out how to start dbus mainloop or attach to the existing dbus session in order to show buttons on the desktop notifications. But it's doable, i.e.: python3-notify2 exists on all major distros since at least ubuntu 16.

Probably I'll add dekstop notifications using libnotify first, to replace systray balloon messages, and then study how to show buttons on notifications in all DEs.

**Comment by gustavo-iniguez-goya** [comment]
This is how it'll look like (with proper formatting):
![image](https://user-images.githubusercontent.com/2742953/148609746-4157d1ec-c337-44fe-8390-f0450320319a.png)

**Comment by GreenLunar** [comment]
Awesome!

What about a "Details" or "More Info" button?
_--- WAIT!_
By all means, the suggested button is good and I'd love to use it just as it is right not

_So..._ What about a "Details" or "More Info" button?
* Assuming the default action is directed by default on executable.
* However, user wants to entirely block the given IP that is displayed on the notification.
* User clicks on "Details" button, and then does the complex action on the popup which we currently have.

**Comment by gustavo-iniguez-goya** [comment]
> What about a "Details" or "More Info" button?

Could be added, but these pop-ups don't offer many options. Out of the box, you can only display text and buttons, so no comboboxes, no checkboxes, no lists, etc. 

The "More info" could display all the fields of the connection, and there could be another button to open the GUI.

**Comment by GreenLunar** [comment]
Of course not.  The "More Info" button will open the popup dialog that we currently have now

**Comment by GreenLunar** [comment]
> This is how it'll look like (with proper formatting): ![image](https://user-images.githubusercontent.com/2742953/148609746-4157d1ec-c337-44fe-8390-f0450320319a.png)

I don't have buttons. I've tested with xfce4 and dunst.

**Comment by gustavo-iniguez-goya** [comment]
Yeah, I haven't added it for this release, only system notifications without buttons. I need to figure out how to show the buttons in all DEs.

**Comment by GreenLunar** [comment]
Buttons are the most crucial part.
Either way, whether you will add it or not, users of other notification software who won't have button, would still, not have buttons.
I suggest to add button, and add note, "Support GNOME, KDE, DE, DE, DE)".
